File: android_flutter_app/lib/models/message.dart
/// Delivery status driving the bubble status icons.
enum MessageStatus { sending, sent, delivered, read }

/// A single chat message.
class Message {
  final String id;
  final String text;
  final bool isMine;
  final DateTime createdAt;
  final MessageStatus status;

  const Message({
    required this.id,
    required this.text,
    required this.isMine,
    required this.createdAt,
    this.status = MessageStatus.sending,
  });

  Message copyWith({MessageStatus? status}) => Message(
        id: id,
        text: text,
        isMine: isMine,
        createdAt: createdAt,
        status: status ?? this.status,
      );

  String get timeLabel =>
      '${createdAt.hour.toString().padLeft(2, '0')}:${createdAt.minute.toString().padLeft(2, '0')}';
}

File: typ0k/Models/Message.swift
import Foundation

enum MessageStatus: String, Codable {
    case sending
    case sent
    case delivered
    case read
    case failed
}

/// Telegram-style inline keyboard button attached to bot messages.
struct InlineButton: Identifiable, Codable, Hashable {
    let id: String
    let title: String
    let action: String
}

struct Message: Identifiable, Codable, Hashable {
    var id: String
    var chatID: String
    var senderID: String
    var text: String
    var attachments: [Attachment] = []
    var createdAt: Date
    var status: MessageStatus = .sending
    var isEdited: Bool = false
    var isDeleted: Bool = false
    var replyToID: String? = nil
    var forwardedFrom: String? = nil
    var reactions: [String: [String]] = [:]
    var inlineButtons: [InlineButton]? = nil
    /// Centered Telegram-gift-style system pill (e.g. "Ваш аккаунт верифицирован").
    var isSystemPill: Bool = false
}

extension Message {
    var isOutgoingTextEmpty: Bool {
        text.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }
}
